plugins/src/webrtcsink/imp.rs

```diff
@@ -1865,17 +1913,23 @@ impl WebRTCSink {
     ) -> Result<gst::Structure, Error> {
         let pipe = PipelineWrapper(gst::Pipeline::new(None));
 
-        let src = match codec.is_video {
-            true => make_element("videotestsrc", None)?,
-            false => make_element("audiotestsrc", None)?,
+        let src = if codec.is_video {
+            make_element("videotestsrc", None)?
+        } else {
+            make_element("audiotestsrc", None)?
         };
-        let capsfilter = make_element("capsfilter", None)?;
+        let mut elements = Vec::new();
+        elements.push(src.clone());
 
-        pipe.0.add_many(&[&src, &capsfilter]).unwrap();
-        src.link(&capsfilter)
-            .with_context(|| format!("Running discovery pipeline for caps {}", caps))?;
+        elements.push(make_converter_for_video_caps(caps)?);
 
-        let (_, _, pay) = setup_encoding(&pipe.0, &capsfilter, codec, None, true)?;
+        let capsfilter = make_element("capsfilter", None)?;
+        elements.push(capsfilter.clone());
+        let elements_slice = &elements.iter().collect::<Vec<_>>();
+        pipe.0.add_many(elements_slice).unwrap();
+        gst::Element::link_many(elements_slice).unwrap();
```

Comment:
don't unwrap here, it shouldn't be fatal, instead keep the `.with_context(|| format!("Running discovery pipeline for caps {}", caps))?;`